File: leetcode-submissions/729_My_Calendar_I.ts
class MyCalendar {
    bookings:{s:number,e:number}[]=[];
    constructor() {
        
    }

    book(start: number, end: number): boolean {
        for(const booking of this.bookings){
            if(Math.max(start,booking.s)<Math.min(booking.e,end)){
                return false;
            }
        }
        this.bookings.push({s:start,e:end});
        return true;
    }
}

/**
 * Your MyCalendar object will be instantiated and called as such:
 * var obj = new MyCalendar()
 * var param_1 = obj.book(start,end)
 */

File: leetcode-submissions/729_My_Calendar_I.cpp
class MyCalendar {
public:
vector<pair<int,int>>bookings;
    MyCalendar() {
        
    }
    
    bool book(int start, int end) {
        for(const auto & [s,e]:bookings ){
            if(max(start,s)<min(end,e)){
                return false;
            }
        }
        bookings.push_back({start,end});
        return true;
    }
};

/**
 * Your MyCalendar object will be instantiated and called as such:
 * MyCalendar* obj = new MyCalendar();
 * bool param_1 = obj->book(start,end);
 */